tests/agent/agent_test.py

```diff
@@ -447,3 +450,65 @@ def process(self, message: agent_message.Message) -> None:
     )
     assert isinstance(logger_error.call_args_list[1][0][1], ValueError) is True
     assert "some error" in logger_error.call_args_list[1][0][1].args[0]
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
+def testAgentAtExist_whenTerminationSignalIsSent_shouldInterceptSignalExecuteAtExistAndExit(
+    agent_run_mock: agent_testing.AgentRunInstance,
+    mocker: plugin.MockerFixture,
+    ping_message: agent_message.Message,
+) -> None:
+    """Ensuring the execution of the `at_exit` method in the case of unexpected agent termination."""
+    internal_q = mp.Queue()
+
+    class TestAgent(agent.Agent):
+        """Helper class to test Agent at exit implementation."""
+
+        def __init__(self, agent_definition, agent_settings, queue) -> None:
+            super().__init__(agent_definition, agent_settings)
+            self.queue = queue
+
+        def process(self, message: agent_message.Message) -> None:
+            del message
+            time.sleep(2000)
+
+        def at_exit(self) -> None:
+            self.queue.put(42)
```

Comment:
```suggestion
            self.queue.put("at_exit called")
```

Reply:
Switching from `mp.Queue` to: `mp.Event(), assert mp_event.is_set() is True`. 
42 is The Answer though!